src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/HttpClientFactoryDisposeTests.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Extensions.Http.Tests
+{
+    public class HttpClientFactoryDisposeTests
+    {
+        [Fact]
+        public async Task DisposingServiceProvider_DisposesHttpClientFactory_ReleasesResources()
+        {
+            // Arrange
+            var disposeCounter = new DisposeCounter();
+            var services = new ServiceCollection();
+            services.AddSingleton(disposeCounter);
+            services.AddTransient<DisposeTrackingHandler>();
+
+            services.AddHttpClient("test")
+                .AddHttpMessageHandler<DisposeTrackingHandler>()
+                .SetHandlerLifetime(TimeSpan.FromMilliseconds(50)); // Very short to ensure quick expiration
```

Comment:
I believe the minimal allowed lifetime is 1s, so this test will most possibly fail.

However, we don't need to wait. In tests, we should be able to "simulate" expiration with `internal virtual void StartHandlerEntryTimer(...)`, `internal virtual void StartCleanupTimer()` etc. -- these methods are marked with comments like "internal for tests". There also should be a test suit here in the repo that uses that. I think it did something like extending "TestHttpClientFactory" from the DefaultHttpClientFactory and allowing access to the methods. See if you can reuse it maybe (but it will need to be extracted to a file then)